ruby/tzdump.rb

```diff
@@ -5,35 +7,50 @@
 START_UTC = DateTime.new(1905, 1, 1, 0, 0, 0, '+0')
 END_UTC = DateTime.new(2035, 1, 1, 0, 0, 0, '+0')
 
+def puts_crlf(*messages)
+  $stdout.write(messages.join("\r\n"))
+  $stdout.write("\r\n")
+end
+
 def format_offset(offset)
   sign = offset < 0 ? "-" : "+"
   offset = offset.abs
   return '%s%02d:%02d:%02d' % [sign, offset / 3600, (offset / 60) % 60, offset % 60]
 end
 
-def dump_zone(console, id)
-  console.write "#{id}\r\n"
+def dump_period(period, zone)
```

Comment:
I think I'll want to rename this - it sounds like it's just dumping a *single* period, whereas actually it's dumping everything for the whole zone.

Also, the fact that it starts with a single period feels odd to me.

How about making it *genuinely* print out the details for a single period (or actually a single transition), and return the next one - then putting the while loop in dump_zone instead?